### PR TITLE
Allow for building plugins separately from VAST

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

In essence, this enables the following workflow:

```bash
cmake -S . -B build.vast
cmake --build build.vast
cmake --install build.vast

cmake -S path/to/plugin -B build.plugin
cmake --build build.plugin
cmake --install build.plugin
```

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Commit-by-commit.

~I still need to make it possible to run unit and integration tests, so while the first commit is reviewable this PR is not complete yet.~

### CODE REVIEW

File: cmake/VASTRegisterPlugin.cmake [hunk 1]
```diff
@@ -81,6 +81,13 @@ function (VASTRegisterPlugin)
     target_link_libraries(${PLUGIN_TARGET}-test PRIVATE vast::test
                                                         vast::internal)
     target_link_whole_archive(${PLUGIN_TARGET}-test PRIVATE ${PLUGIN_TARGET})
+    add_test(NAME build-${PLUGIN_TARGET}-test
+              COMMAND "${CMAKE_COMMAND}" --build "${CMAKE_BINARY_DIR}"
+                      --config "$<CONFIG>" --target ${PLUGIN_TARGET}-test)
+    set_tests_properties(build-${PLUGIN_TARGET}-test
+      PROPERTIES FIXTURES_SETUP vast_unit_test_fixture)
```
Comment (tobim) [inline cmake/VASTRegisterPlugin.cmake]: check if this is really required?

File: examples/plugins/example/CMakeLists.txt [hunk 2]
```diff
@@ -4,9 +4,10 @@ project(example
   DESCRIPTION "example plugin for VAST"
   LANGUAGES CXX)
 
-if (NOT TARGET vast::libvast)
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
```
Comment (tobim) [inline examples/plugins/example/CMakeLists.txt]: Please add a comment:
```
If the plugin is built in stand-alone mode.
```

File: examples/plugins/example/CMakeLists.txt [hunk 3]
```diff
@@ -4,9 +4,10 @@ project(example
...
+  include(CTest)
```
Comment (tobim) [inline examples/plugins/example/CMakeLists.txt:8]: Please explain why this is necessary here.

File: cmake/VASTRegisterPlugin.cmake (hunk 1 above)
Comment (dominiklohmann) [reply cmake/VASTRegisterPlugin.cmake]: It is so it runs first.

**Review by tobim — APPROVED** [review]
I verified that this works locally.

**Comment by dominiklohmann** [comment]
I need to rebase this once more after #1444 is in, then I'll merge it.

**Pull request merged**